### DISCUSSION TITLE
Username and Password authentication

### DISCUSSION THREAD

**Discussion by TkachouAndrej** [pnp/pnpcore#598] [Q&A] [answered]
Hi, 
could you please provide me with an example where we connect to Sharepoint
with the user name and the password only, without MFA.

I tried to use PnPCoreAuthenticationUsernamePasswordOptions, like below:

services.AddPnPCoreAuthentication(
                 options =>
               {
                  options.Credentials.Configurations.Add(_sharePointAuthenticationOptions.AuthenticationProvider,
                         new PnP.Core.Auth.Services.Builder.Configuration.PnPCoreAuthenticationUsernamePasswordOptions
                         {
                           Username = _sharePointAuthenticationOptions.Username,
                           Password = _sharePointAuthenticationOptions.Password
                         });
               });

but I have error:

Argument 2: cannot convert from 'PnP.Core.Auth.Services.Builder.Configuration.PnPCoreAuthenticationUsernamePasswordOptions' 
to 'PnP.Core.Auth.Services.Builder.Configuration.PnPCoreAuthenticationCredentialConfigurationOptions' 

What should I use instead of this?
I only have found examples where X509 certificates and ClientId+TenantId were used.

Thank you

**Comment by jansenbe** [comment, marked as answer]
Hi @TkachouAndrej , this sample (https://pnp.github.io/pnpcore/demos/Demo.RPi/README.html) shows how to use the `UsernamePasswordAuthenticationProvider`. This only works when there's no MFA, like you mentioned. Also do know that you need to have an Azure AD app setup.
